### ISSUE TITLE
How to use it in Idea?

### ISSUE THREAD

**Issue by winhkey** [apache/maven-mvnd#548]
Hi:
It is a good tool. but how to use it in Idea? now i can only run in terminal :D

**Comment by gnodet** [comment]
My understanding is that Intellij Idea uses a compilation server by default and does not delegate to maven (only when generating sources very specific build steps), so I don't think there's much to gain here.

**Comment by yxxcrtd** [comment]
Direct input：mvnd clean in IDEA's terminal

**Comment by ppalaga** [comment]
> how to use it in Idea?

What exactly would you expect?

**Comment by krasa** [comment]
Perhaps you would find this plugin useful, but only just as an alternative to manually running goals in a terminal (https://github.com/krasa/MavenHelper/issues/85)
- https://plugins.jetbrains.com/plugin/7179-maven-helper/

Proper support must be provided by IntelliJ itself - https://youtrack.jetbrains.com/issue/IDEA-264396

**Comment by zhaoyunxing92** [comment]
I feel like I can add it like this before I run it

![image](https://user-images.githubusercontent.com/18088210/148537829-444e00b2-a6fc-4787-a101-514f3895da4e.png)

**Comment by winhkey** [comment]
> Perhaps you would find this plugin useful, but only just as an alternative to manually running goals in a terminal ([krasa/MavenHelper#85](https://github.com/krasa/MavenHelper/issues/85))
> 
> * https://plugins.jetbrains.com/plugin/7179-maven-helper/
> 
> Proper support must be provided by IntelliJ itself - https://youtrack.jetbrains.com/issue/IDEA-264396

Yes, I have been using maven helper, but it is not support for mvnd, so I made this issue :)

**Comment by winhkey** [comment]
> I feel like I can add it like this before I run it
> 
> ![image](https://user-images.githubusercontent.com/18088210/148537829-444e00b2-a6fc-4787-a101-514f3895da4e.png)

good idea~

**Comment by fprochazka** [comment]
This is relevant if you want to always build your project using maven and not by the IDEA mechanism - we have a bunch of plugins configured there and its beneficial to be able to build locally the same way its build on CI. 

I was able to make it work using a custom bash script, that is invoked as a run configuration, but the problem is that IDEA terminal is doing something with the output (I have Ubuntu) and run that takes 3sec takes about 10sec to render so you have to wait for it to catch up to show you if it was successful. I would expect a native support to work more effectively with IDEA.

**Comment by krasa** [comment]
@fprochazka Sounds like IntelliJ problem. https://intellij-support.jetbrains.com/hc/en-us/articles/207241235-Reporting-performance-problems

**Comment by fprochazka** [comment]
@krasa I agree, but still, having IDEA plugin that would allow using mvnd instead of mvn/mvnw would be great.

**Comment by fprochazka** [comment]
Yup. I've upvoted about every related issue in their tracker :D but some of them are years old, so it probably won't get fixed any time soon.
